### DISCUSSION TITLE
[BE201]安裝 nginx 後訪問遠端的 ip 沒有出現 welcome to nginx(已解決)

### DISCUSSION THREAD

**Discussion by KW0409** [Lidemy/forum#230] [後端相關問題] [answered]
安裝完 nginx 之後，訪問遠端的 ip 出現的不是 welcome to nginx 的頁面，而是出現 ubuntu 的預設頁面，非常神奇
<img width="70%" alt="截圖 2021-11-01 下午8 02 17" src="https://user-images.githubusercontent.com/80152099/139666270-557bdecb-8ef7-430b-9fe1-70dc33ec7522.png">

在網上有找到一篇情況跟我類似的文章 [安装Nginx后，访问IP看不到Welcome to Nginx！- CSDN](https://blog.csdn.net/weixin_44914711/article/details/109381995)
但是按照他的方式，找到 /usr/share/nginx/html 目录，並沒有發現裡面的 index.html 有指向別的檔案
(打開這個 index.html 發現裡面的內容也確實是 welcome to nginx 頁面，不知道那個網友的 terminal 會有顯示指到別的檔案是不是因為有安裝其他的套件)
<img width="70%" alt="截圖 2021-11-01 下午8 02 17" src="https://user-images.githubusercontent.com/80152099/139666817-3c2cc9d2-63e5-4cfb-afd0-cab8dc5355c5.png">

後來在 /etc/nginx/sites-available 裡面看到了 default 文件，看起來很像配置文件，不知道是不是配置有什麼問題
<img width="70%" alt="截圖 2021-11-01 下午8 02 17" src="https://user-images.githubusercontent.com/80152099/139668494-ff65ad54-c173-4eea-9873-247d3f9c940f.png">

P.S.
有確認過 ufw status, nginx status 都是正常的
(ufw status 的 Nginx Full 是因為看了這篇 [Ubuntu 20.04 伺服器設定 (Nginx, PHP, MySQL, phpMyAdmin, Python 3.9)](https://the.annswer.org/t/topic/354)，也有試過課程影片裡的 Nginx HTTP 結果都是一樣的)
<img width="70%" alt="截圖 2021-11-01 下午8 02 17" src="https://user-images.githubusercontent.com/80152099/139669071-dec4058b-d6c6-4279-be06-73028e861b6b.png">
<img width="70%" alt="截圖 2021-11-01 下午8 02 17" src="https://user-images.githubusercontent.com/80152099/139669074-be10462a-fb73-4d6b-9504-a4476a8e1c8c.png">

**Comment by ddylanlinn** [comment]
關掉阿帕契 並試著看 80 port 被誰占用!? `sudo netstat -tulpn`

**Reply by aszx87410** [reply]
對，可以先看一下 80 port 是不是 nginx

**Reply by KW0409** [reply]
阿啊抱歉，忘了說 80 port 也有先關掉 apache2 並看過確實是 nginx 在使用沒錯🥲
<img width="70%" alt="截圖 2021-11-01 下午9 23 12" src="https://user-images.githubusercontent.com/80152099/139678761-779dfff0-4b60-4b4c-b7b0-81bc11cb5512.png">

**Comment by aszx87410** [comment, marked as answer]
你的 nginx config 裡面看起來讀檔案的根目錄是 /var/www/html，看起來跟 apache 的一樣，所以會讀到 apache 的 index.html
要驗證的話，你試著在這個資料夾底下建一個 test.html 然後看能不能連到就知道了

**Reply by KW0409** [reply]
剛剛在 `/var/www/html` 裡面有找到另一個檔案 index.nginx-debian.html
裡面的內容就是 nginx 的預設畫面
測試了一下能正常連到 index.nginx-debian.html，這樣應該是代表 nginx 確實沒問題吧🥺
<img width="70%" alt="截圖 2021-11-01 下午10 06 33" src="https://user-images.githubusercontent.com/80152099/139685008-d5dfb5bb-9f71-4f82-b101-848f8b45408b.png">

我本來還以爲這個檔案會這樣命名是因為用 nginx 連線的時候會是開啟這個檔案，現在看來應該都是開 index.html，
這樣是不是表示如果是先安裝 apache 再安裝 nginx 的，初始畫面都會是 apache 的畫面啊

**Reply by aszx87410** [reply]
這樣應該就代表沒問題沒錯

**Reply by KW0409** [reply]
好的！感謝 Huli 老師
